app/javascript/groceries/components/CheckInItem.vue

```diff
@@ -65,19 +64,15 @@ function aboutToMoveToClass() {
 }
 
 function moveTo(checkInStatus: CheckInStatus) {
-  if (props.item.aboutToMoveTo) return;
```

Comment:
This line was causing a bug if e.g. I would check on item as being in cart and then rapidly uncheck it. With this line, it would remain in the "In Cart" section (though it would become unchecked). With this change, it moves back to the "Needed" section, as expected.